### PR TITLE
Init: coqPackages.vcfloat at v2.1.1

### LINKED ISSUE #257548: Package request: CoqPackages.vcfloat
**Project description**

VCFloat: A Unified Coq Framework for Verifying C Programs with Floating-Point Computations

**Metadata**

* homepage URL: https://verinum.org/ 
* source URL: https://github.com/VeriNum/vcfloat
* license: LGPL
* platforms: <uncertain limitations here>

### PR DESCRIPTION
**Pull request by quinn-dougherty**
## Description of changes
Closes https://github.com/NixOS/nixpkgs/issues/257548

VCFloat: A Unified Coq Framework for Verifying C Programs with Floating-Point Computations

Metadata
    - homepage URL: https://verinum.org/
    - source URL: https://github.com/VeriNum/vcfloat
    - license: LGPL

## Things done

Built on `x86_64-linux`

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
Please read the [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) and in particular follow the [commit conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions), thanks.

File: pkgs/development/coq-modules/vcfloat/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  release."2.1.1".sha256 =
+    "sha256-bd/XSQhyFUAnSm2bhZEZBWB6l4/Ptlm9JrWu6w9BOpw=";
+  releaseRev = v: "v${v}";
+  propagatedBuildInputs = [ interval compcert flocq bignums ];
+  meta = {
+    description =
+      "VCFloat is a tool for Coq proofs about floating-point round-off error.";
+    maintainers = with maintainers; [ quinn-dougherty ];
+  };
+}
```
Comment (Zimmi48) [inline pkgs/development/coq-modules/vcfloat/default.nix:23]: Is this auto-formatted with some kind of auto-formatter? I would prefer if you could follow a closer formatting style to what is currently done for Coq packages (see e.g., the example from the [documentation](https://nixos.org/manual/nixpkgs/unstable/#sec-language-coq)). In particular, make the `defaultVersion` more compact, add empty lines to give some space (that makes reading easier), do not add newlines between `=` and strings (release hash and description).

This is also missing a `license` field.

Comment (quinn-dougherty) [reply pkgs/development/coq-modules/vcfloat/default.nix:23]: thanks! I was careful to disable format on save for preexisting files, didn't know it was a priority for new files. Will cover for next commit.

**Comment by mweinelt** [comment]
I'm locking and closing this pull request, because through the rebase you inadvertently requested many maintainers for review, which subscribed them to update notifications, resulting in unnecessary spam inside everyone's inbox. While we can remove their review requests, we sadly cannot unsubscribe anyone.

Please create a new pull request and for the next time remember to set your PR to draft status before rebasing. In draft status, you can preview the list of maintainers that are about to be requested for review, which allows you to sidestep this issue.

**Pull request closed without merging**